### PR TITLE
Arm64 work

### PR DESCRIPTION
**Pull request by schultetwin1**
### Description of the changes:
- Adds support for CPU detection by CMake
- Removes SSE instructions from ARM builds
- Adds CMake ARM toolchains files
- Updates the docker image to support cross compiling

### Before submitting a Pull Request:
- [x] I reviewed [CONTRIBUTING.md](https://github.com/Microsoft/Azure-Kinect-Sensor-SDK/blob/develop/CONTRIBUTING.md)
- [x] I [built my changes](https://github.com/Microsoft/Azure-Kinect-Sensor-SDK/blob/develop/docs/building.md) locally
- [x] I ran the [unit tests](https://github.com/Microsoft/Azure-Kinect-Sensor-SDK/blob/develop/docs/testing.md)
- [ ] I ran the [functional tests](https://github.com/Microsoft/Azure-Kinect-Sensor-SDK/blob/develop/docs/testing.md) with a device
- [ ] I ran the [performance tests](https://github.com/Microsoft/Azure-Kinect-Sensor-SDK/blob/develop/docs/testing.md) with a device

### I tested changes on:
- [ ] Windows
- [x] Linux

This is currently a draft to get our CI system up and running.

### CODE REVIEW

File: src/transformation/rgbz.c [hunk 1]
```diff
@@ -8,9 +8,12 @@
 #include <limits.h>
 #include <math.h>
 
+#if defined(__amd64__) || defined(_M_AMD64) || defined(__i386__) || defined(_M_X86)
```
Comment (wes-b) [inline src/transformation/rgbz.c]:
```suggestion
#ifdef __SSE4_1__
```

File: src/transformation/rgbz.c [hunk 2]
```diff
@@ -1055,42 +1058,45 @@ k4a_buffer_result_t transformation_color_image_to_depth_camera_internal(
     return K4A_BUFFER_RESULT_SUCCEEDED;
 }
 
+#if !defined(K4A_USING_SSE)
```
Comment (wes-b) [inline src/transformation/rgbz.c]:
```suggestion
#ifndef __SSE4_1__
```

File: src/transformation/rgbz.c (hunk 1 above)
Comment (schultetwin1) [reply src/transformation/rgbz.c]: Does that work for MSVC as well?

From here it looks like it does not:
https://docs.microsoft.com/en-us/cpp/preprocessor/predefined-macros?view=vs-2019

Comment (wes-b) [reply src/transformation/rgbz.c]: no

**Comment by wes-b** [comment]
Work has been completed on a feature branch instead of this branch

**Pull request closed without merging**